syntax = "proto3";

message AuthRequest {
  string uid = 1;
  string credentials = 2;
}

message AuthResponse {
  string data = 1;
}

message Text {
  string data = 1;
}

message User {
  string uid = 1;
}

message Dialog {
  string name = 1;
  string dialog_id = 2;
  string date = 3;
  string message = 4;
}

message Dialogs {
  repeated Dialog dialog = 1;
}

message Send {
  string uid = 1;
  string entity = 2;
  string message = 3;
}

message StatusMessage {
  string status = 1;
}

message Chunk {
  bytes chunk = 1;
}

service GmailApi {
  rpc auth(AuthRequest) returns (AuthResponse) {};
  rpc get_dialogs(User) returns (Dialogs) {};
  rpc get_messages(Text) returns (Text) {};
  rpc send_message(Send) returns (StatusMessage) {};
}
